syntax = "proto3";

package grpc;

message Empty {}

message Str {
    string sender = 1;
    string message = 2;
    string recipient = 3;
}

message Account {
    string username = 1;
    bool created = 2;
    bool loggedIn = 3;
}

service ChatServer {
    // This bi-directional stream makes it possible to send and receive Notes between 2 persons
    rpc ChatStream (Account) returns (stream Str);
    rpc createAccount (Account) returns (Account);
    rpc deleteAccount (Account) returns (Str);
    rpc listAccounts (Empty) returns (Str);
    rpc login(Account) returns (Account);
    rpc sendStr (Str) returns (Str); 
    rpc dequeue (Account) returns (Str);
}
